[AXC-3476] Re-enable excluded arm64 architecture

josiprezic-mohemian opened the pull request:
## 🎫 Ticket

- https://airside.atlassian.net/browse/AXC-3476

## 👀 Related pull requests

- https://github.com/airsidemobile/nfc-sdk-ios/pull/155
- https://github.com/airsidemobile/habicht-ios/pull/2336

## 📖 Description

The exclusion of `arm64` for the iPhone simulator was likely introduced to address compatibility issues with certain dependencies in the past. However, as all dependencies now fully support the `arm64` architecture ([as mentioned here](https://github.com/AndyQ/NFCPassportReader/pull/230)), this exclusion is no longer required.

Given recent issues and complications arising from this configuration in the [nfc-sdk-ios](https://github.com/airsidemobile/nfc-sdk-ios), along with the need to update the CI pipeline to use Xcode 16 instead of Xcode 15, I have removed this exclusion.

mikahautz-mohemian reviewed (approved):
LGTM ✅

Just curious, what is the issue with this configuration that you mentioned for the nfc-sdk?

josiprezic-mohemian commented:
> Just curious, what is the issue with this configuration that you mentioned for the nfc-sdk?

@mikahautz-mohemian So the changes from this pull request fixed this specific issue in the nfc-sdk project related CI-jobs:

`
Testing failed:
        xctest (16202) encountered an error (Failed to load the test bundle. (Underlying Error: The bundle “NFCSDKTests” couldn’t be loaded because it doesn’t contain a version for the current architecture. The bundle doesn’t contain a version for the current architecture. Try installing a universal version of the bundle. dlopen_preflight(/Users/distiller/project/DerivedData/Build/Products/Beta-iphonesimulator/NFCSDKTests.xctest/NFCSDKTests) => false, tried: '/Users/distiller/project/DerivedData/Build/Products/Beta-iphonesimulator/NFCSDKTests' (no such file), '/Applications/Xcode-16.2.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib/NFCSDKTests' (no such file), '/Library/Developer/CoreSimulator/Volumes/iOS_22C150/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 18.2.simruntime/Contents/Resources/RuntimeRoot/Users/distiller/project/DerivedData/Build/Products/Beta-iphonesimulator/NFCSDKTests.xctest/NFCSDKTests' (no such file), '/Users/distiller/project/DerivedData/Build/Products/Beta-iphonesimulator/NFCSDKTests.xctest/NFCSDKTests' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))))
`

---

You may ask what caused this issue in the first place, so here is a short timeline:

1. Upgraded from Xcode 15 to Xcode 16.
2. The `arch` parameter in the `xcodebuild` command was no longer supported in the same way as in Xcode 15, causing build failures.
3. To address this, I updated the `Scanfile` by removing `run_rosetta_simulator` config entry.
4. After this change, tests failed with an error about incompatible architecture mentioned above.
4. To resolve this, I updated `NFCPassportReader` to stop excluding the `arm64` configuration, as this exclusion was no longer required.
5. CI job finally succeeded 🎉

mikahautz-mohemian commented:
> You may ask what caused this issue in the first place, so here is a short timeline:
> 
> 1. Upgraded from Xcode 15 to Xcode 16.
> 2. The `arch` parameter in the `xcodebuild` command was no longer supported in the same way as in Xcode 15, causing build failures.
> 3. To address this, I updated the `Scanfile` by removing `run_rosetta_simulator` config entry.
> 4. After this change, tests failed with an error about incompatible architecture mentioned above.
> 5. To resolve this, I updated `NFCPassportReader` to stop excluding the `arm64` configuration, as this exclusion was no longer required.
> 6. CI job finally succeeded 🎉

I see, makes sense! Thanks for the explanation @josiprezic-mohemian! 🙌
